test/integration/history/test_spark_history_server.py

```diff
@@ -90,3 +100,22 @@ def _request_with_retry(url, max_retries=10):
         return http.request("GET", url)
     except Exception:  # pylint: disable=W0703
         return None
+
+
+# due to s3 eventual consistency, s3 file may not be uploaded when we kick off history server
+def _wait_for_file_to_be_uploaded(region, bucket, prefix, file_name, file_size, timeout=20):
+    s3_client = boto3.client("s3", region_name=region)
+
+    max_time = time.time() + timeout
+    while True:
+        response = s3_client.list_objects(Bucket=bucket, Prefix=prefix)
+        if "Contents" in response:
+            for file in response["Contents"]:
+                if file["Key"] == f"{prefix}/{file_name}" and file["Size"] == file_size:
+                    print(f"{file_name} was fully uploaded.")
+                    return
+
+        if time.time() > max_time:
+            raise TimeoutError("Timeout waiting for s3 upload to be consistent")
+
+        time.sleep(1)
```

Comment:
Suggestion, not necessary: you may want to use the `smspark.waiter` instead, which has the same logic.

Reply:
Yeah, I think I'll leave this as it is, since I find that I need to define another function like 
```
def method():
    _wait_for_file_to_be_uploaded(region, bucket, spark_event_logs_key_prefix, file_name, file_size)
```
and then 
```
waiter.wait_for(predicate_fn=method, timeout=20, period=1)
```

Or do you know if there's a way to pass function with arguments to wait_for() directly?